app/routes/recommend_routes.py

```diff
@@ -0,0 +1,92 @@
+import psycopg2
+import psycopg2.extras
+from dotenv import load_dotenv
+import os
+
+load_dotenv()
+
+recommend_routes = Blueprint('recommend_routes', __name__)
+
+logger = get_logger("RecoEndpoint")
+
+DB_CONFIG = {
+    "dbname": os.getenv("DB_NAME"),
+    "user": os.getenv("DB_USER"),
+    "password": os.getenv("DB_PASSWORD"),
+    "host": os.getenv("DB_HOST"),
+    "port": int(os.getenv("DB_PORT")),
+}
+
+@recommend_routes.route("/recommend", methods=["POST"])
+def recommend():
+    """
+    Handles a recommendation request based on the user's movie preferences.
+
+    1. Extracts the IDs of the movies liked by the user from the database.
+    2. Loads the movie similarity model.
+    3. Generates a list of recommended movies based on the user's liked movies and the similarity matrix.
+
+    :return: A JSON response with the recommended movies or an error message.
+        - In case of success, returns a list of recommended movies.
+        - In case of failure, returns an error message in JSON format.
+    """
+    logger.info("Received a recommendation request")
+    saver = MovieSaver()
+    trainer = MovieTrainer()
+
+    try:
+        with psycopg2.connect(**DB_CONFIG) as conn:
+            logger.info("Connected to the database")
+            with conn.cursor(cursor_factory=psycopg2.extras.DictCursor) as cursor:
+                query = "SELECT movie_id FROM user_favorite_movies WHERE user_id = %s"
+                user_id = request.json.get("user_id")
+                if not user_id:
+                    logger.warning(f"User ID {user_id} not provided in the request")
+                    return jsonify({"message": "User ID is required"}), 400
+
+                logger.info(f"Fetching liked movies for user_id: {user_id}")
+                cursor.execute(query, (user_id,))
+                liked_movie_ids = [row["movie_id"] for row in cursor.fetchall()]
+
+                logger.debug(f"Liked movie IDs: {liked_movie_ids}")
+
+                if not liked_movie_ids:
+                    logger.warning(f"No liked movies found for user_id: {user_id}")
+                    return jsonify({"message": "No liked movies found for the user"}), 404
+
+        movies_df = trainer.load_movies_from_db()
+        model_path = "./movie_similarity_model.pkl"
+        loaded_similarity_matrix = saver.load_model(model_path)
+
+        if movies_df is None:
+            logger.error("Movies dataframe is None")
+            return jsonify({"message": "Movie data not found"}), 500
+        if loaded_similarity_matrix is None:
+            logger.error("Similarity matrix is None")
+            return jsonify({"message": "Model not found"}), 500
+
+        logger.info("Generating movie recommendations")
+        recommender = MovieRecommender(movies_df)
+        recommendations = recommender.generate_recommendations(liked_movie_ids, movies_df, loaded_similarity_matrix)
+
+        logger.info("Displaying top 20 recommendations.")
```

Comment:
we don't need top 20 recommendations here
calculate the mean similarity score between the liked movies and each candidate movie
return all results in the request formatted as
```
[
    {
        "movie_id": 1,
        "score": 0.85
    },
    {
        "movie_id": 2,
        "score": 0.75
    }
]
```